pkg/archive/archive.go

```diff
@@ -1143,6 +1144,35 @@ loop:
 	return nil
 }
 
+// createImpliedDirectories will create all parent directories of the current path with default permissions, if they do
+// not already exist. This is possible as the tar format supports 'implicit' directories, where their existence is
+// defined by the paths of files in the tar, but there are no header entries for the directories themselves, and thus
+// we most both create them and choose metadata like permissions.
+//
+// The caller should have performed filepath.Clean(hdr.Name), so hdr.Name will now be in the filepath format for the OS
+// on which the daemon is running. This precondition is required because this function assumes a OS-specific path
+// separator when checking that a path is not the root.
+func createImpliedDirectories(dest string, hdr *tar.Header, options *TarOptions) error {
```

Comment:
Bike-shedding: perhaps `ensureParentDirectories()` would be a good alternative name 😅.

Reply:
I actually originally used that name, but I found this one to be more explicit in terms of intent (it's meant very specifically for 'missing'/'implied' directories in a tar layout). Not a hill I need to die on, but if you don't have strong opinions I do prefer this name (it's more symmetrical with the `const`).

Reply:
No strong attachment. The "create implied" was _slightly_ harder to grasp (but very much because I don't use "implied" often in my vocabulary, and I'm a non-native English speaker).